mifosng-provider/src/main/java/org/mifosplatform/portfolio/loanaccount/service/LoanAssembler.java

```diff
@@ -172,8 +177,38 @@ private Loan assembleApplication(final JsonElement element, final Long clientId,
         if (loanPurposeId != null) {
             loanPurpose = this.codeValueRepository.findOneWithNotFoundDetection(loanPurposeId);
         }
+        Set<LoanDisbursementDetails> disbursementDetails = null;
+        BigDecimal fixedEmiAmount = null;
+        if (loanProduct.isMultiDisburseLoan() || loanProduct.canDefineInstallmentAmount()) {
+            fixedEmiAmount = this.fromApiJsonHelper.extractBigDecimalWithLocaleNamed(LoanApiConstants.emiAmountParameterName, element);
+        }
+        BigDecimal maxOutstandingLoanBalance = null;
+        if (loanProduct.isMultiDisburseLoan()) {
+            disbursementDetails = fetchDisbursementData(element.getAsJsonObject());
+            final Locale locale = this.fromApiJsonHelper.extractLocaleParameter(element.getAsJsonObject());
+            maxOutstandingLoanBalance = this.fromApiJsonHelper.extractBigDecimalNamed(LoanApiConstants.maxOutstandingBalanceParameterName,
+                    element, locale);
+            if (disbursementDetails.isEmpty()) {
+                final String errorMessage = "For this loan product, disbursement details must be provided";
+                throw new MultiDisbursementDataRequiredException(LoanApiConstants.disbursementDataParameterName, errorMessage);
+            }
+
+            if (disbursementDetails.size() > loanProduct.maxTrancheCount()) {
+                final String errorMessage = "Number of tranche shouldn't be greter than " + loanProduct.maxTrancheCount();
+                throw new ExceedingTrancheCountException(LoanApiConstants.disbursementDataParameterName, errorMessage,
+                        loanProduct.maxTrancheCount(), disbursementDetails.size());
+            }
+        }
+        Set<LoanCharge> loanCharges = new HashSet<LoanCharge>();
+        Set<LoanCharge> loanTrancheCharges = null;
+        Set<LoanCharge> charges = null;
+        if(loanProduct.isTrancheChargeExist()){
+            loanTrancheCharges = createChargesFromProductDefinition(loanProduct,element,disbursementDetails);
+        }
+        charges = this.loanChargeAssembler.fromParsedJson(element);
+        loanCharges.addAll(loanTrancheCharges);
```

Comment:
This would throw a null pointer exception for loans which do not have disbursement charges